include/swift/AST/AutoDiff.h

```diff
@@ -251,54 +249,36 @@ struct SILAutoDiffConfig {
 };
 
 /// The kind of an associated function in the `autodiff_function` and
-/// `autodiff_extract` instructions in SIL.
-enum class SILAutoDiffAssociatedFunctionKind {
-  // The primal function in legacy reverse-mode.
-  LegacyPrimal,
-  // The adjoint function in legacy reverse-mode.
-  LegacyAdjoint,
-  // The vector-Jacobian products operator.
-  JVP,
-  // The differential, the result of JVP except that it has boxed arguments
-  // (closure captures) for partial application within JVP.
-  Differential,
-  // The Jacobian-vector products operator.
-  VJP,
-  // The pullback, the result of JVP except that it has boxed arguments
-  // (closure captures) for partial application within VJP.
-  Pullback
+/// `autodiff_function_extract` instructions in SIL.
+struct AutoDiffAssociatedFunctionKind {
+  enum innerty : uint8_t {
+     // The vector-Jacobian products operator.
```

Comment:
Are the comments correct? The words are in the opposite order of the letters of the enum values.